core/adapters/index.ts

```diff
@@ -41,6 +41,10 @@ export enum WarehouseType {
   SQLDATAWAREHOUSE = "sqldatawarehouse"
 }
 
+export function isWarehouseType(input: any): input is WarehouseType {
```

Comment:
`input: string`?

Reply:
It may take string, since this does type narrowing using user-defined type guard, I think it would be better to take any or unknown.